test/anchoredinterval.jl

```diff
@@ -148,141 +148,253 @@ using Intervals: canonicalize
+                ),
+                string(
+                    "AnchoredInterval{-1 day,$(tz_prefix)ZonedDateTime}",
+                    "(2016-08-12T00:00:00-05:00, Inclusivity(false, true))"
+                ),
+            ),
+            (
+                AnchoredInterval{Minute(-5)}(dt),
+                "(2016-08-11 5ME02:00]",
+                string(
+                    "AnchoredInterval{-5 minutes,DateTime}",
+                    "(DateTime(2016, 8, 11, 2, 0, 0), Inclusivity(false, true))",
+                ),
+                string(
+                    "AnchoredInterval{-5 minutes,DateTime}",
+                    "(2016-08-11T02:00:00, Inclusivity(false, true))",
+                ),
+            ),
+            (
+                AnchoredInterval{Second(-30)}(dt),
+                "(2016-08-11 30SE02:00:00]",
+                string(
+                    "AnchoredInterval{-30 seconds,DateTime}",
+                    "(DateTime(2016, 8, 11, 2, 0, 0), Inclusivity(false, true))",
+                ),
+                string(
+                    "AnchoredInterval{-30 seconds,DateTime}",
+                    "(2016-08-11T02:00:00, Inclusivity(false, true))",
+                ),
+            ),
+            (
+                AnchoredInterval{Millisecond(-10)}(dt),
+                "(2016-08-11 10msE02:00:00.000]",
+                string(
+                    "AnchoredInterval{-10 milliseconds,DateTime}",
+                    "(DateTime(2016, 8, 11, 2, 0, 0), Inclusivity(false, true))",
+                ),
+                string(
+                    "AnchoredInterval{-10 milliseconds,DateTime}",
+                    "(2016-08-11T02:00:00, Inclusivity(false, true))",
+                ),
+            ),
+        ]
 
-        interval = AnchoredInterval{Second(-30)}(dt)
-        @test string(interval) == "(2016-08-11 30SE02:00:00]"
-        @test sprint(show, interval, context=:compact=>true) == string(interval)
-        @test sprint(show, interval) == mod_prefix * string(
-            "AnchoredInterval{-30 seconds,DateTime}(2016-08-11T02:00:00, ",
-            "Inclusivity(false, true))",
-        )
 
-        interval = AnchoredInterval{Millisecond(-10)}(dt)
-        @test string(interval) == "(2016-08-11 10msE02:00:00.000]"
-        @test sprint(show, interval, context=:compact=>true) == string(interval)
-        @test sprint(show, interval) == mod_prefix * string(
-            "AnchoredInterval{-10 milliseconds,DateTime}(2016-08-11T02:00:00, ",
-            "Inclusivity(false, true))",
-        )
+        for test in display_tests
+            interval = test[1]
+            @test string(interval) == test[2]
+            @test sprint(show, interval, context=:compact=>true) == string(interval)
+            if VERSION < v"1.2.0-DEV.29"
+                @test sprint(show, interval) == mod_prefix * test[4]
+            else
+                @test sprint(show, interval) == mod_prefix * test[3]
+            end
+        end
```

Comment:
We should add tests for `repr` and `sprint(print, ...)` so there behaviour is well defined